### PR TITLE
[GoogleCloudSpannerReceiver] Spanner fgac user

### PR DESCRIPTION
**Pull request by harishbohara11**
**Description:** <Describe what has changed.>
Spanner introduced [Fine-grained access control](https://cloud.google.com/spanner/docs/fgac-about) which allows use of database roles for accessing a database. Added an optional field for specifying the database role associated with the database.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>
Manual testing with local docker setup done. Unit tests were added for the new config change.

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Review by songy23 — COMMENTED** [review]
Please add changelog

**Comment by harishbohara11** [comment]
Added the changelog.

File: .chloggen/spanner_fgac_user.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: breaking
```
Comment (MovieStoreGuy) [inline .chloggen/spanner_fgac_user.yaml:4]: Could you explain how this becomes a breaking change?

File: receiver/googlecloudspannerreceiver/config.go [hunk 2]
```diff
@@ -95,10 +100,18 @@ func (instance Instance) Validate() error {
...
 
 	return nil
 }
+
+func (database Database) Validate() error {
+	if database.Name == "" {
+		return errors.New("field \"databases\" contains empty database names")
+	}
+
+	return nil
```
Comment (MovieStoreGuy) [inline receiver/googlecloudspannerreceiver/config.go:116]: Should there also be a role defined here?

**Review by MovieStoreGuy — COMMENTED** [review]
These changes make sense, I am curious if this will change existing behaviour though.

**Comment by MovieStoreGuy** [comment]
Could I ask you to fix up the linting issues please :)

**Comment by KiranmayiB** [comment]
I dont seem to be an assigned reviewer. Please let me know what I should do? @harishbohara11

File: .chloggen/spanner_fgac_user.yaml (hunk 1 above)
Comment (harishbohara11) [reply .chloggen/spanner_fgac_user.yaml:4]: This PR changes the way databases are mentioned in the config file.

Earlier:
```
databases:
            - "db11"
            - "db12"
```

Now:
```
databases:
            - database_name: "db11"
              fgac_database_role: "database_role"
            - database_name: "db12"`
```

File: receiver/googlecloudspannerreceiver/config.go (hunk 2 above)
Comment (harishbohara11) [reply receiver/googlecloudspannerreceiver/config.go:116]: Empty roles are allowed. Roles are required only when FGAC role is being used to access the DB.

**Comment by harishbohara11** [comment]
@MovieStoreGuy Thanks for reviewing. Fixed linting issues.
@KiranmayiB here is the codeowner for cloud spanner receiver. Requesting her to have a look.

**Comment by KiranmayiB** [comment]
LGTM

**Review by songy23 — APPROVED** [review]

**Comment by jpkrohling** [comment]
@MovieStoreGuy, do you think your concerns were properly addressed?

**Comment by maduris** [comment]
hi @MovieStoreGuy , @songy23, @jpkrohling,

We are also looking at the spanner receiver but with FGAC and bumped into an issue where FGAC is not supported by the current spanner receiver. We noticed this PR was raised and approved but did not manage to get merged. Wondering if there is anyway you could reopen this PR and merge it or would this require another PR?

cc: @harishbohara11 

Regards
Sai

**Pull request closed without merging**